src/main/java/app/linguistai/bmvp/response/gamification/RUserXPRanking.java

```diff
@@ -0,0 +1,16 @@
+package app.linguistai.bmvp.response.gamification;
+
+import app.linguistai.bmvp.model.User;
+import app.linguistai.bmvp.model.gamification.UserXP;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+@Builder
+@Data
+@AllArgsConstructor
+public class RUserXPRanking {
+    private User user;
```

Comment:
Since `User` class has the `password` field, using it in the response classes can also return the `password of the User` (unwanted behavior). Can you check if this is the case?

If so, you can use some existing response class or add a new response class for the `User` class that only contains relevant details (`email`, `UUID`, `username` etc.).

Reply:
User's password field has @JsonIgnore annotation so their password is not returned (only id, username, email).